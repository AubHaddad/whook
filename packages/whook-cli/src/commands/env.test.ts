import { describe, it, beforeEach, jest, expect } from '@jest/globals';
import initEnvCommand from './env.js';
import { YError } from 'yerror';
import type { LogService } from 'common-services';
import type { PromptArgs } from '../services/promptArgs.js';

describe('envCommand', () => {
  const promptArgs = jest.fn<PromptArgs>();
  const log = jest.fn<LogService>();

  beforeEach(() => {
    log.mockReset();
    promptArgs.mockReset();
  });

  it('should work', async () => {
    promptArgs.mockResolvedValueOnce({
      command: 'whook',
      rest: ['env'],
      namedArguments: {
        name: 'NODE_ENV',
      },
    });

    const envCommand = await initEnvCommand({
      log,
      ENV: { NODE_ENV: 'test' },
      promptArgs,
    });
    const result = await envCommand();

    expect({
      result,
      promptArgsCalls: promptArgs.mock.calls,
      logCalls: log.mock.calls.filter(([type]) => !type.endsWith('stack')),
    }).toMatchSnapshot();
  });

  it('should work with a default value', async () => {
    promptArgs.mockResolvedValueOnce({
      command: 'whook',
      rest: ['env'],
      namedArguments: {
        name: 'NODE_ENV',
        default: 'lol',
      },
    });

    const envCommand = await initEnvCommand({
      log,
      ENV: {},
      promptArgs,
    });
    const result = await envCommand();

    expect({
      result,
      promptArgsCalls: promptArgs.mock.calls,
      logCalls: log.mock.calls.filter(([type]) => !type.endsWith('stack')),
    }).toMatchSnapshot();
  });

  it('should fail with no value', async () => {
    promptArgs.mockResolvedValueOnce({
      command: 'whook',
      rest: ['env'],
      namedArguments: {
        name: 'NODE_ENV',
      },
    });

    const envCommand = await initEnvCommand({
      log,
      ENV: {},
      promptArgs,
    });

    try {
      await envCommand();
      throw new YError('E_UNEXPECTED_SUCCESS');
    } catch (err) {
      expect({
        errorCode: (err as YError).code,
        errorParams: (err as YError).params,
        promptArgsCalls: promptArgs.mock.calls,
        logCalls: log.mock.calls.filter(([type]) => !type.endsWith('stack')),
      }).toMatchSnapshot();
    }
  });
});
