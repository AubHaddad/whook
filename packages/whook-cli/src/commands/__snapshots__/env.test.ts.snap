// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`envCommand should fail with no value 1`] = `
{
  "errorCode": "E_NO_ENV_VALUE",
  "errorParams": [
    "NODE_ENV",
  ],
  "logCalls": [],
  "promptArgsCalls": [
    [],
  ],
}
`;

exports[`envCommand should work 1`] = `
{
  "logCalls": [
    [
      "info",
      "test",
    ],
  ],
  "promptArgsCalls": [
    [],
  ],
  "result": undefined,
}
`;

exports[`envCommand should work with a default value 1`] = `
{
  "logCalls": [
    [
      "info",
      "lol",
    ],
  ],
  "promptArgsCalls": [
    [],
  ],
  "result": undefined,
}
`;
